nepomuk/src/main/java/edu/asu/diging/gilesecosystem/nepomuk/core/service/handlers/AbstractFileHandler.java

```diff
@@ -104,5 +104,15 @@ public String getRelativePathInTypeFolder(IFile file) {
     
     protected abstract IFileStorageManager getStorageManager();
     
+    public void deleteFile(IFile file, boolean isOldFileVersion) {
+        if(!isOldFileVersion) {
```

Comment:
what is this check for?

Reply:
When we reprocess a document the file in the same path is overwritten. So there are multiple oldFileVersionIds or 1 oldFileVersionId and the current fileID 

- The file will be deleted the first time the request comes
- The next time the request comes again for another oldFileID or the current fileID the file will not be present and an exception will be thrown.

Thus I skipped deleting from storage for old file IDs and only delete the file database entry.
And when the current file ID is received the file is deleted from storage as well.